submissions/kt-std/memory-pair-game/js/script.js

```diff
@@ -0,0 +1,145 @@
+	numCards = 9,
+	cardsData = ((assetsPath) => {
+		const arr = [];
+		for (let i = 1; i <= numCards; i++) {
+			const card = {path: `${assetsPath}p${i}.svg`, name: `p${i}`};
+			arr.push(card, card);
+		}
+		return arr;
+	})("assets/");
+
+let checkedCards = [],
+	checkedCardsContainers = [],
+	shuffledCards = shuffleArray(cardsData),
+	foundMatches = [];
+
+function shuffleArray(o) {
+	for (
+		var j, x, i = o.length;
+		i;
+		j = parseInt(Math.random() * i), x = o[--i], o[i] = o[j], o[j] = x
+	);
+	return o;
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+	appendCards(shuffledCards);
+});
+
+function appendCards(shuffledCards) {
+	const fragment = document.createDocumentFragment();
+	shuffledCards.forEach((card, index) => {
+		fragment.appendChild(createCard(card.name, card.path, index));
+	});
+	container.appendChild(fragment);
+}
+
+function createCard(cardName, cardImgPath, index) {
+	const card = document.createElement("div"),
+		cardContainer = document.createElement("div"),
+		cardBack = document.createElement("div"),
+		cardFront = document.createElement("div"),
+		img = document.createElement("img");
+	img.src = cardImgPath;
+	img.alt = `${cardName} card image`;
+	img.classList.add("card__image");
+	cardBack.classList.add("card_back");
+	cardBack.addEventListener("click", checkCard);
+	cardFront.appendChild(img);
+	cardFront.classList.add("card_front", "flex-center");
+	cardFront.setAttribute("name", cardName);
+	cardContainer.classList.add("card__container");
+	cardContainer.appendChild(cardBack);
+	cardContainer.appendChild(cardFront);
+	card.appendChild(cardContainer);
+	card.classList.add("card");
+	setOpacity(cardContainer, 1, index * 150);
+	return card;
+}
+
+function checkCard(e) {
+	if (checkedCards.length < 2) {
+		const cardContainer = e.target.offsetParent;
+		cardContainer.classList.add("flip");
+		checkedCards.push(e.target.nextSibling.attributes.name.value);
+		checkedCardsContainers.push(cardContainer);
+		if (checkedCards.length == 2) {
+			if (checkedCards[0] === checkedCards[1]) {
+				foundMatches = foundMatches.concat(checkedCards);
+				checkedCardsContainers.forEach((el) => {
+					changeSize(el);
+				});
+			} else {
+				checkedCardsContainers.forEach((el) => {
+					flipBack(el);
+				});
+			}
+			setTimeout(() => {
+				checkedCardsContainers = [];
+				checkedCards = [];
+			}, 600);
+		}
+	}
+	if (foundMatches.length === shuffledCards.length) {
+		const winContainer = document.querySelector(".win__container");
+		if (winContainer !== null) {
+			winContainer.style.opacity = "0";
+			winContainer.style.display = "flex";
+			setOpacity(winContainer, 1);
+		} else {
+			showWinMessage();
+		}
+	}
+}
+
+function changeSize(card) {
+	setTimeout(() => {
+		card.style.width = "0";
+		card.style.height = "0";
+		card.childNodes.forEach(
+			(cardInnerStyle) => (cardInnerStyle.style.display = "none")
+		);
+	}, 500);
+}
+
+function setOpacity(property, value, delay) {
+	if (delay === undefined) {
```

Comment:
I would return the `return` value. Mutating function parameter is really confusing.

Reply:
Do we actually need to return it? This if statement is only used to set the default value of the delay if this parameter was omitted during the function call.

Reply:
I think you are looking for [default parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)

Reply:
@AMashoshyna, thank you for your time and helpful advises ! I've fixed it with the default parameter.